service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gcsbucket/ControlledGcsBucketResource.java

```diff
@@ -214,7 +215,13 @@ public ControlledGcsBucketResource.Builder common(ControlledResourceFields commo
     }
 
     public ControlledGcsBucketResource.Builder bucketName(@Nullable String bucketName) {
-      this.bucketName = Optional.ofNullable(bucketName).orElse(generateBucketName());
+      if (StringUtils.isEmpty(bucketName)) {
```

Comment:
can you also move this to the callsite and here we just do this.bucketName = bucketName. and remove nullable.